src/plugins/intel_gpu/include/intel_gpu/graph/network.hpp

```diff
@@ -213,6 +214,13 @@ struct network {
     /// Returns memory state @p variable_id of stateful network
     VariableState& get_variable_memory(const std::string &variable_id);
 
+    /// Add APIs for handling kernels_cache
+    void compile();
+    kernel_id add_kernel(const std::shared_ptr<kernel_string>& kernel_sring);
+    kernel::ptr get_kernel(kernel_id id);
```

Comment:
I think we can mark this method as `const`

Reply:
I will change it. thanks.

Reply:
Remove these APIs because they are not necessary any more after refactoring codes.